smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructureGenerator.java

```diff
@@ -158,6 +158,7 @@ private void renderErrorStructure() {
 
         writer.openBlock("export interface $L extends $L {", symbol.getName(), extendsFrom);
         writer.write("__type: $S;", shape.getId().getName());
+        writer.write("name: $S;", shape.getId().getName());
```

Comment:
I thought we were just special casing exceptions. Why is this needed on all structures? Why would we need this *and* __type?

Reply:
This is in the code for exception structures: `renderErrorStructure`

Reply:
Ok. This needs to be addressed in a followup PR.

Reply:
Issue created for removing __type https://github.com/awslabs/smithy-typescript/issues/96